File: utils/dataLoader.js
// helpers/dataLoader.js

/**
 * datLoader.js
 *
 * Utility for loading test data (CSV or JSON) for Playwright GUI tests.
 *
 * Equivalent Python version: utils/data_loader.py
 * Both use the shared data files under: tests/data/
 */


const fs = require('fs');
const path = require('path');
const { parse } = require('csv-parse/sync');

function loadTestData(fileName) {
  const fullPath = path.resolve(__dirname, '../tests/data', fileName);

  if (fileName.endsWith('.json')) {
    return JSON.parse(fs.readFileSync(fullPath, 'utf-8'));
  } else if (fileName.endsWith('.csv')) {
    const csvContent = fs.readFileSync(fullPath, 'utf-8');
    const records = parse(csvContent, {
      columns: true,
      skip_empty_lines: true,
    });

    return records.map(row => ({
      ...row,
      should_pass: row.should_pass.trim().toLowerCase() === 'true',
    }));
  } else {
    throw new Error(`Unsupported file format: ${fileName}`);
  }
}

module.exports = { loadTestData };

File: utils/data_loader.py
# utils/data_loader.py

import json
import csv
import os

def get_project_root():
    return os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))

def load_test_data(filename):
    """
    Load test data from either the 'tests' or 'data' directory.
    Use subdir="data" for shared CSV/JSON test input.
    """

    data_dir = os.path.join(get_project_root(), 'tests', 'data')
    full_path = os.path.join(data_dir, filename)

    if filename.endswith(".json"):
        with open(full_path, 'r') as f:
            return json.load(f)
    elif filename.endswith(".csv"):
        with open(full_path, newline='') as f:
            reader = csv.DictReader(f)
            return list(reader)
    else:
        raise ValueError(f"Unsupported file type: {filename}")

